### DISCUSSION TITLE
Add Webserver Library with uPyCraft on ESP32?

### DISCUSSION THREAD

**Discussion by Cripacx** [micropython/micropython#10514] [General]
Hello,
i want to add a webserver library on the ESP32 using uPyCraft. For example tinyweb or MicroWebSrv2. 
How do i add a library like them correctly? I only found docs about how to add it to a raspberry.

Greetings

**Comment by jimmo** [comment]
I don't know about uPyCraft (which I thought had been abandoned a few years ago) or tinyweb or microwebsrv2.

However https://github.com/miguelgrinberg/microdot is very easy to use, and can be installed just by copying the relevant files to the device. (e.g. using mpremote or Thonny).

**Reply by jimmo** [reply]
(Also, we are working on finding a way to add microdot to `mip`, and then eventually apply this to other libraries)

**Comment by jczic** [comment]
Hi @Cripacx, I'm the author of MicroWebSrv 1&2.
If you have an ESP32, you also try my IDE on GitHub : [ESP32-MPY-Jama](https://github.com/jczic/ESP32-MPY-Jama)
It's new so if you have feedback, I take!

**Reply by dannyelectro** [reply]
this looks very good. As a novice programmer it is what I am looking for. Can't wait to try it later :)
